handlers/users.go

```diff
@@ -0,0 +1,73 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"tweeter/db/models/user"
+	"tweeter/handlers/responses"
+)
+
+// UsersHandler is the /api/v1/users/ endpoint
+func UsersHandler(w http.ResponseWriter, req *http.Request) {
+	if req.Method == http.MethodPut {
+		handleUserCreate(w, req)
+	} else {
+		renderErrors(w, http.StatusBadRequest, responses.Error{
+			Title: "Unsupported Method", Detail: fmt.Sprintf("Method %s is not supported (yet)", req.Method),
+		})
+	}
+}
+
+func parseID(w http.ResponseWriter, req *http.Request) (ID user.ID, ok bool) {
+	idString := strings.TrimSuffix(strings.TrimPrefix(req.URL.Path, "/api/v1/users/"), "/")
+	id, err := user.ParseID(idString)
+	if err != nil {
+		renderErrors(w, http.StatusBadRequest, responses.Error{
+			Title: "Invalid User ID", Detail: fmt.Sprintf("Failed to parse ID from %s, err: %s", idString, err),
+		})
+		return 0, false
+	}
+
+	return id, true
+}
+
+func handleUserCreate(w http.ResponseWriter, req *http.Request) {
+	body, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		renderErrors(w, http.StatusBadRequest, responses.Error{
+			Title: "Malformed Body", Detail: fmt.Sprintf("Failed to read request body, err: %s", err),
```

Comment:
Probably no need to return the internal error to the user, let's just log it (and switch to a logging framework later).